lib/async_hooks.js

```diff
@@ -476,6 +476,31 @@ function init(asyncId, type, triggerAsyncId, resource) {
   processing_hook = false;
 }
 
+// Integration with V8 Inspector AsyncTask* API
+
+const inspectorHook = createHook({
+  init(asyncId, type, triggerId, resource) {
+    // TODO(bajtos): support recurring tasks like setInterval
+    const recurring = false;
+    async_wrap.asyncTaskScheduled(type, asyncId, recurring);
+  },
+
+  before(asyncId) {
+    async_wrap.asyncTaskStarted(asyncId);
+  },
+
+  after(asyncId) {
+    async_wrap.asyncTaskFinished(asyncId);
+  },
+
+  destroy(asyncId) {
+    async_wrap.asyncTaskCanceled(asyncId);
+  },
+});
+
+// TODO(bajtos): enable the hook when Inspector is started,
+// disable it when Inspector is stopped
+inspectorHook.enable();
```

Comment:
For accessing the hook; you could set up 2 persistent Function properties on the Environment (like the async_hooks_blah functions in env.h), then add a binding method to the inspector module/inspector_agent.cc that takes two parameters and sets the fields on the Environment accordingly.

One of these functions would be basically `() => hook.enable()`, the other `() => hook.disable()`. You could then grab those methods in `Agent::Start` or `Agent::Stop` off the Environment instance.